### ISSUE TITLE
Cannot load umd module (need add module object in eval)

### ISSUE THREAD

**Issue by unlight** [Paciolan/remote-module-loader#5]
UMD (rollup) wrapper could looks like this:
```
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
    typeof define === 'function' && define.amd ? define('h-document-element', ['exports'], factory) :
    (global = global || self, factory(global['h-document-element'] = {}));
}(this, function (exports) { 'use strict';
```
So, checking for `exports` is not enough, need add `module` in this eval
```
var exports = {};
new Function("require", "exports", data)(requires, exports);
```

Webpack umd:
```
(function webpackUniversalModuleDefinition(root, factory) {
	if(typeof exports === 'object' && typeof module === 'object')
		module.exports = factory();
	else if(typeof define === 'function' && define.amd)
		define([], factory);
	else {
		var a = factory();
		for(var i in a) (typeof exports === 'object' ? exports : root)[i] = a[i];
	}
})(window, function() {
```

**Comment by unlight** [comment]
@joelnet will you fix it?

**Comment by joelnet** [comment]
I have updated this to `2.3.0` which should now support UMD modules.

Can you please test this and verify that it is working for you?

**Comment by unlight** [comment]
Does not work.
Issue about remote loaded modules, but not this module `@paciolan/remote-module-loader`.
`createLoadRemoteModule` function should be changed to something like this:
```
const module = { exports };
new Function('require', 'module', 'exports', code)(requires, module, exports);
return module.exports;
```

**Comment by joelnet** [comment]
Hmm. I have added the babel plugin `@babel/plugin-transform-modules-umd`, which should have handled that.

Do you have some example code that I could use to test this?

**Comment by unlight** [comment]
Sure, here is the example.
```js
import { createLoadRemoteModule} from '@paciolan/remote-module-loader';

test('umd load', async () => {
    const remoteModuleLoader = createLoadRemoteModule();
    const result = await remoteModuleLoader('https://unpkg.com/h-document-element@2.0.0/bundles/h-document-element.umd.js');
    expect(result.Fragment).toBeDefined();
    expect(result.createElement).toBeDefined();
    expect(result.h).toBeDefined();
});
```

**Comment by joelnet** [comment]
AHH Got it. Misunderstanding. I made `remove-module-loader` accessible as a UMD. You were referring to loading a UMD.

Now that is cleared up, I can fix the right thing. lol

**Comment by joelnet** [comment]
A fix has been pushed to 2.4.0

**Comment by unlight** [comment]
d3adef2429da26cbe14b08a895d9e581b79be104 It should fix. Thank you.
